src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ISZMetricsCollectorSpark.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SamPairUtil;
+import htsjdk.samtools.metrics.MetricsFile;
+import org.apache.spark.api.java.JavaRDD;
+import org.broadinstitute.hellbender.metrics.MetricAccumulationLevel;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * Worker class to collect insert size metrics.
+ */
+public class ISZMetricsCollectorSpark implements Serializable{
+    private static final long serialVersionUID = 1L;
+
+    // uses TreeMap because it is serializable and sorted
+    private TreeMap<Integer, Long> hist;
+
+    public ISZMetricsCollectorSpark(final JavaRDD<GATKRead> filteredReads,
+                                    final Set<MetricAccumulationLevel> accumLevels,
+                                    final List<SAMReadGroupRecord> samRgRecords,
+                                    final double devTolerance) {
+
+        final Map<Integer, Long> toBeCasted = filteredReads.map(read -> Math.abs(read.getFragmentLength()) ).countByValue();
```

Comment:
inline this temporary variable and mention that you're making a copy to ensure sorted order